cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -0,0 +1,361 @@
+                        horizontalDatum, publicName, longName, description, timezoneId, countyName, stateInitial, active, locationKindId,
+                        mapLabel, publishedLatitude, publishedLongitude, boundingOfficeId, nationId, nearestCity, aliasResultSet);
+                Location location = new Location.Builder(locationId[0], locationKindId[0], ZoneId.of(timezoneId[0]), latitude[0].doubleValue(), longitude[0].doubleValue(), horizontalDatum[0], officeId)
+                        .withLocationType(locationType[0])
+                        .withElevation(elevation[0].doubleValue())
+                        .withVerticalDatum(verticalDatum[0])
+                        .withPublicName(publicName[0])
+                        .withLongName(longName[0])
+                        .withDescription(description[0])
+                        .withCountyName(countyName[0])
+                        .withStateInitial(stateInitial[0])
+                        .withActive(active[0])
+                        .withMapLabel(mapLabel[0])
+                        .withBoundingOfficeId(boundingOfficeId[0])
+                        .withNearestCity(nearestCity[0])
+                        .withNation(Nation.NationForName(nationId[0]))
+                        .build();
+                if (publishedLatitude[0] != null) {
+                    location = new Location.Builder(location)
+                            .withPublishedLatitude(publishedLatitude[0].doubleValue())
+                            .build();
+                }
+                if (publishedLongitude[0] != null) {
+                    location = new Location.Builder(location)
+                            .withPublishedLongitude(publishedLongitude[0].doubleValue())
+                            .build();
+                }
+                locationRef.set(location);
+            });
+        }
+        catch(DataAccessException ex)
+        {
+            throw new IOException("Could not retrieve Location");
+        }
+        return locationRef.get();
+    }
+
+    @Override
+    public void deleteLocation(String locationName, String officeId) throws IOException
+    {
+        try
+        {
+            dsl.connection(c ->
+            {
+                CwmsDbLoc locJooq = CwmsDbServiceLookup.buildCwmsDb(CwmsDbLoc.class, c);
+                locJooq.delete(c, officeId, locationName);
+            });
+        }
+        catch(DataAccessException ex)
+        {
+            throw new IOException(ex);
+        }
+    }
+
+    @Override
+    public void storeLocation(Location location) throws IOException
+    {
+        try
+        {
+            dsl.connection(c ->
+            {
+                CwmsDbLoc locJooq = CwmsDbServiceLookup.buildCwmsDb(CwmsDbLoc.class, c);
+                String elevationUnits = "m"; //this should be updated to use Unit enum once merged in
```

Comment:
I like what this comment says, what stopping it from getting done now before merging in?

Reply:
Wrote this code before we merged in the Unit enum. Updated it to use `Unit.METER.getValue()`